### ISSUE TITLE
"smart" dashes don't work with beamer

### ISSUE THREAD

**Issue by pbreheny** [jgm/pandoc#4763]
Suppose my input file `test.md` looks like

```
This sentence's problem is -- a dash.
```

When I run `pandoc test.md -f markdown -t latex`, everything works fine:

```
This sentence's problem is -- a dash.
```

When I run `pandoc test.md -f markdown -t beamer`, however, I get

```
\begin{frame}

This sentence’s problem is – a dash.

\end{frame}
```

Two problems: the apostrophe is curvy and the double hyphen has become an en-dash, both of which will cause problems if I try to compile the latex document.

I can run `pandoc test.md -f markdown-smart -t beamer`, which is a little better:

```
\begin{frame}

This sentence's problem is -\/- a dash.

\end{frame}
```

The curvy apostrophe is gone, but I now have `-\/-` instead of `--`, which will render as two hyphens instead of an en-dash.

* Is this really the intended behavior?  I can't see how it could be.
* Is there any way to get an en-dash in the beamer output?

I am using Pandoc v 2.1.2.

**Comment by mb21** [comment]
try `-t beamer+smart`

**Comment by pbreheny** [comment]
Ah, OK, everything makes sense now.  My misunderstanding on how `-smart` and `+smart` worked in Pandoc 2+.  For anyone else who might be running into the same issue, `pandoc test.md -f markdown -t beamer+smart` works perfectly.  Thanks for your help!

**Comment by jgm** [comment]
+smart shouldn't be needed; that was a bug that has
been fixed in version 2.2.  But that workaround will
work if you're still on 2.1.x.

**Comment by pbreheny** [comment]
Just upgraded to pandoc 2.2.1 and indeed, everything works perfectly with the default `pandoc file.md -t beamer` now.  Thank you both so much.
